modin/experimental/sql/__init__.py

```diff
@@ -12,15 +12,64 @@
 # governing permissions and limitations under the License.
 
 import warnings
+import modin.pandas as pd
+import modin.config as cfg
 
-try:
-    from dfsql import sql_query as query
-    import dfsql.extensions  # noqa: F401
-except ImportError:
-    warnings.warn(
-        "Modin experimental sql interface requires dfsql to be installed."
-        + " Run `pip install modin[sql]` to install it."
-    )
-    raise
-
-__all__ = ["query"]
+_query_impl = None
+
+
+def query(sql, *args, **kwargs):
+    """
+    Execute SQL query using either hdk_query or dfsql.
+
+    Parameters
+    ----------
+    sql : str
+        SQL query to be executed.
+    *args : tuple
+        Positional arguments, passed to the execution engine.
+    **kwargs : dict
+        Keyword arguments, passed to the execution engine.
+
+    Returns
+    -------
+    modin.pandas.DataFrame
+        Execution result.
+    """
+    global _query_impl
+
+    if _query_impl is None:
+        if cfg.StorageFormat.get() == "Hdk":
+            from modin.experimental.core.execution.native.implementations.hdk_on_native import (
+                hdk_query,
+            )
+
+            _query_impl = hdk_query
+        else:
+            try:
+                from dfsql import sql_query
+                import dfsql.extensions  # noqa: F401
+
+                _query_impl = sql_query
+            except ImportError:
+                warnings.warn(
+                    "Modin experimental sql interface requires dfsql to be installed."
+                    + " Run `pip install modin[sql]` to install it."
+                )
+                raise
+
+    return _query_impl(sql, *args, **kwargs)
+
+
+# Dfsql adds the slq() method to the DataFrame class.
```

Comment:
```suggestion
# dfsql adds the slq() method to the DataFrame class.
```

Reply:
Btw, why do we need this code below? Isn't it enough the code in query method?

Reply:
dfsql has an extension, that injects the sql() method into the DataFrame class - https://github.com/mindsdb/dfsql/blob/dbe5979f2e9b79de196cc17143b66b11e06851ba/dfsql/extensions.py

import dfsql.extensions in query() is redundant, I forgot to remove it.